common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java

```diff
@@ -247,24 +237,22 @@ public static void runTests(PerfTestBase<?>[] tests, boolean sync, int parallel,
 
         try {
             if (sync) {
-                ForkJoinPool forkJoinPool = new ForkJoinPool(parallel);
-                List<Callable<Integer>> operations = new ArrayList<>(parallel);
-                for (PerfTestBase<?> test : tests) {
-                    operations.add(() -> {
-                        test.runAll(endNanoTime);
-                        return 1;
-                    });
-                }
-
-                forkJoinPool.invokeAll(operations);
-
-                // Sleep until the tests complete.
-                Thread.sleep(durationSeconds * 1000L);
+                if (parallel > 1) {
+                    ForkJoinPool forkJoinPool = new ForkJoinPool(parallel);
+                    List<Callable<Integer>> operations = new ArrayList<>(parallel);
+                    for (PerfTestBase<?> test : tests) {
+                        operations.add(() -> {
+                            test.runAll(endNanoTime);
+                            return 1;
+                        });
+                    }
 
-                forkJoinPool.shutdown();
+                    forkJoinPool.invokeAll(operations);
 
-                // Wait 10 seconds for operations to shut down.
-                forkJoinPool.awaitTermination(10, TimeUnit.SECONDS);
+                    forkJoinPool.awaitQuiescence(durationSeconds + 1, TimeUnit.SECONDS);
+                } else {
```

Comment:
Why do we need to special-case parallel=1?  I'd prefer to use the same codepath for paralell=1 and parallel>1 if possible.

Reply:
I'll revert this